src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -92,42 +101,7 @@ public void storeMessage(Message message) {
    *    there are no messages
    */
   public List<Message> getAllMessages() {
-    return messageQuery(defaultMessageUser);
-  }
-
-  /**
-   * Helper function to add versatility to redundant code for Message Queries.
-   *
-   * @return a list of messages p
-   */
-  private List<Message> messageQuery(String user) {
-    List<Message> messages = new ArrayList<>();
-
-    Query query = (user != null) ? new Query("Message")
-                                    .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
-                                    .addSort("timestamp", SortDirection.DESCENDING) :
-                                  new Query("Message")
-                                    .addSort("timestamp", SortDirection.DESCENDING);
-    PreparedQuery results = datastore.prepare(query);
-
-    for (Entity entity : results.asIterable()) {
-      try {
-        String idString = entity.getKey().getName();
-        UUID id = UUID.fromString(idString);
-        String text = (String) entity.getProperty("text");
-        long timestamp = (long) entity.getProperty("timestamp");
-        String recipient = (String) entity.getProperty("recipient");
-
-        Message message = new Message(id, user, text, timestamp, recipient);
-        messages.add(message);
-      } catch (Exception e) {
-        System.err.println("Error reading message.");
-        System.err.println(entity.toString());
-        e.printStackTrace();
-      }
-    }
-
-    return messages;
+    return getMessages(defaultMessageUser, "user");
```

Comment:
Whenever typing in strings that are used for comparison throughout a class I've found using a constant helps not only for readability but also reducing typos.

For example: I would assign `"user"` to a global constant called `USER_QUERY` and replace all instances of `"user"` with `USER_QUERY`, with something similar done for `"recipient"` below.

Reply:
changed to global USER and RECIPIENT constants